src/OrchardCore/OrchardCore.Infrastructure/Secrets/Stores/ConfigurationSecretStore.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.Secrets.Models;
+
+namespace OrchardCore.Secrets.Stores;
+
+public class ConfigurationSecretStore : ISecretStore
+{
+    private readonly IConfigurationSection _configuration;
+    protected readonly IStringLocalizer S;
+
+    public ConfigurationSecretStore(
+        IShellConfiguration shellConfiguration,
+        IStringLocalizer<ConfigurationSecretStore> stringLocalizer)
+    {
+        _configuration = shellConfiguration.GetSection("OrchardCore_Secrets:Secrets");
+        S = stringLocalizer;
+    }
+
+    public string Name => nameof(ConfigurationSecretStore);
+    public string DisplayName => S["Configuration Secret Store"];
+    public bool IsReadOnly => true;
+
+    public Task<SecretBase> GetSecretAsync(string name, Type type)
+    {
+        if (!typeof(SecretBase).IsAssignableFrom(type))
```

Comment:
we need to check whether it is possible to (accidently) save a key, like your email password, to the database store even though it was originally provided via a configuration store.

My memory says we don't allow this to occur, I thi9nk we can display the secret (but not it's value), if hte store is readonly.